File: kafka/golang/routes/route.go
package routes

import (
	"log"
	"note-golang-kafka/controllers"
	"note-golang-kafka/services"

	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
	"github.com/labstack/echo/v4"
)

func SetKafkaRoute(e *echo.Echo) {
	config := &kafka.ConfigMap{
		"bootstrap.servers": "localhost:9092",
	}
	producer, err := kafka.NewProducer(config)
	if err != nil {
		log.Fatalln("error creating producer:", err)
	}
	kafkaService := services.NewKafkaService(producer)
	kafkaController := controllers.NewKafkaController(kafkaService)
	e.GET("/api/v1/message/send-message", kafkaController.SendMessage)
}

File: kafka/rust/src/routes/route.rs
use std::sync::Arc;

use axum::{routing::get, Router};

use crate::{controllers::kafka_controller::send_message, producers::kafka_producer::KafkaProducerImpl, services::kafka_service::KafkaServiceImpl, states::app_state::AppState};

pub fn set_route(kafka_producer: Arc<KafkaProducerImpl>) -> Router {
    let kafka_service = Arc::new(KafkaServiceImpl::new(kafka_producer));
    let state = Arc::new(AppState{kafka_service: kafka_service.clone()});
    Router::new()
        .route("/kafka/send-message/{message}", get(send_message))
        .with_state(state)
}